src/EFCore.PG/Query/Sql/Internal/NpgsqlQuerySqlGenerator.cs

```diff
@@ -198,6 +209,17 @@ protected override Expression VisitBinary(BinaryExpression expression)
             }
         }
 
+        /// <inheritdoc />
+        protected override Expression VisitDefault(DefaultExpression e)
+        {
+            Sql.Append(DefaultConstants.GetOrAdd(e.Type, GenerateLiteralFromMapping));
+            return e;
+
+            string GenerateLiteralFromMapping(Type t)
+                => TypeMappingSource.FindMapping(t)
+                                    .GenerateSqlLiteral(t.IsNullableType() ? null : Activator.CreateInstance(t));
```

Comment:
Why not a lambda expression?

Reply:
Having this inline led to either a really long line....or some odd looking indentation (even by my standards :)).

I thought this looked a _tiny bit_ more readable, but just barely. I don't have a strong opinion on this one, so I can refactor if you think the lambda is more readable.

Reply:
What do you think about the following code?

```csharp
Sql.Append(DefaultConstants.GetOrAdd(e.Type, t=>
    TypeMappingSource.FindMapping(t).GenerateSqlLiteral(
        t.IsNullableType() ? null : Activator.CreateInstance(t)));
```